core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -779,6 +788,31 @@ void handleSqlBasicCallForTimestampMulti(SqlWriter writer, SqlCall call) {
     writer.print(literalValue.toString());
   }
 
+  private static String removeSingleQuotes(SqlNode sqlNode) {
+    return sqlNode.toString().replaceAll("'", "");
+  }
+
+  private void unparseDateTrunc(SqlWriter writer, SqlCall call) {
+    final SqlWriter.Frame dateTruncFrame = writer.startFunCall(call.getOperator().getName());
+    call.operand(1).unparse(writer, 0, 0);
+    writer.print(",");
+    int len = call.operand(0).toString().length();
+    switch (call.operand(0).toString().substring(1, len - 1).toUpperCase(Locale.ROOT)) {
```

Comment:
Extract "call.operand(0).toString().substring(1, len - 1).toUpperCase(Locale.ROOT)" to a method